src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -93,10 +96,10 @@ public void register(Router router) {
     router.put(RELATIVE_ITEMS_PATH_ID).handler(this::update);
     router.delete(RELATIVE_ITEMS_PATH_ID).handler(this::deleteById);
 
-    router.post(RELATIVE_ITEMS_PATH_ID + "/mark-in-process")
-      .handler(handle(this::markAsInProcess));
-    router.post(RELATIVE_ITEMS_PATH_ID + "/mark-in-process-non-requestable")
-      .handler(handle(this::markAsInProcessNonRequestable));
+    Arrays.stream(ItemStatusName.values()).
+      filter(itemStatusName -> ItemStatusURL.getUrlForItemStatusName(itemStatusName).isPresent())
+      .forEach(itemStatusName -> registerHandlerForChangingItemStatus(itemStatusName,router));
```

Comment:
Why has the structure of this changed so that it is necessary to call `ItemStatusURL.getUrlFor` twice?

Personally, I think this is less clear than the transformation, filter then register. Did my suggestions cause this confusion?

Reply:
Is it will be OK?
    Arrays.stream(ItemStatusName.values()).map(itemStatusName -> ItemStatusURL.getUrlForItemStatusName(itemStatusName))
      .filter(itemStatusUrl -> itemStatusUrl.isPresent())
      .forEach(itemStatusUrl -> registerHandlerForChangingItemStatusUrl(itemStatusUrl.get(),router));

  private void registerHandlerForChangingItemStatusUrl(String itemStatusUrl, Router router) {
    router.post(RELATIVE_ITEMS_PATH_ID + itemStatusUrl)
      .handler(handle(this::markItemAsTargetStatus));
  }

or may be so:
registerHandlerForItemStatuses(router);

private void registerHandlerForItemStatuses(Router router) {
    Arrays.stream(ItemStatusName.values())
      .map(itemStatusName -> ItemStatusURL.getUrlForItemStatusName(itemStatusName))
      .filter(itemStatusUrl -> itemStatusUrl.isPresent())
      .forEach(itemStatusUrl -> router.post(RELATIVE_ITEMS_PATH_ID + itemStatusUrl)
        .handler(handle(this::markItemAsTargetStatus)));
  }

Reply:
Yes, I think both are ok. 

I prefer the first option. Having thought more about the language used in this area, I wonder if a better name for the last method might be `registerMarkAsHandler`. What do you think.

If you choose to take the second option, then the `.handler` needs to be indented because it is a continuation of the `router.post` line rather than part of the stream processing chain.

Reply:
RegisterItemStatusTransitionUrlAndHandler - is it precision description of action. May be you, as native speaker, could  interpret it somehow better.

Reply:
I think the name should be consistent with the domain language we have chosen, which is to precede each status with `mark as`. 

I think that means using a name like `registerMarkItemAsHandler`. This gives it a degree of symmetry with the handler method name `markItemAsTargetStatus`. If `registerMarkItemAsTargetStatus` works better for you, I'm ok with that too.

Reply:
let's finish on "registerMarkItemAsHandler".